app/controllers/users_controller.rb

```diff
@@ -0,0 +1,78 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :load_user, only: [:show, :edit, :destroy]
+
+  def current_user? user
+    user == current_user
+  end
+
+  def show
+  end
+
+  def index
+    @users = User.paginate(page: params[:page])
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def destroy
+    if @user.destroy
+       flash[:success] = t "user_deleted"
+    else
+      flash[:error] = t "manipulation_fail"
+    redirect_to users_url
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      redirect_to @user
+      flash[:success] = t "Welcome_to_the_sample_app!"
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @user.update(user_params)
+      flash[:success] =  t"update"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def load_user
+    return if @user = User.find_by(id: params[:id])
+    flash[:warning] = t ".not_found"
+    redirect_to user_path
+  end
+
+  def user_params
+    params.require(:user).permit :name, :email, :password,
+                                 :password_confirmation
+  end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t"please_log_in"
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    redirect_to(root_url)
```

Comment:
logic hàm này lạ thế, sao chuyển trang rồi mới check `@user == current_user`
```suggestion
    return if @user == current_user
    
    flash[:danger] = t "..."
    redirect_to(root_url)
```